### PR TITLE
Remove placeholder

### LINKED ISSUE #244: Remove all custody and VDF placeholders
### What is wrong?

See https://github.com/ethereum/eth2.0-specs/pull/488

### How can it be fixed

Fill this in if you know how to fix it.

### PR DESCRIPTION
**Pull request by hwwhww**
### What was wrong?

Fix #244

### How was it fixed?
- Update the type of `ValidatorIndex` from `uint24` to `uint64`
- Update `SlashableAttestation` and `verify_slashable_attestation` logic.
    - Modify `verify_slashable_attestation` (spec: https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#verify_slashable_attestation) to `validate_slashable_attestation` - it raises `ValidatationError` when it's invalid
    - Implement `verify_bitfield`(spec: https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#verify_bitfield) and change it to `validate_bitfield` - it raises `ValidatationError` when it's invalid
- Remove the placeholders
    - Remove `latest_vdf_outputs`, `persistent_committees`, `persistent_committee_reassignments`, and `custody_challenges` from `BeaconState`
    - Remove `custody_reseeds`, `custody_challenges`,  and `custody_responses` from `BaseBeaconBlock`
    - Remove `custody_commitment` from `ValidatorRecord` and `DepositInput`

[//]: # (For important changes, please add a new entry to the release notes file)
[//]: # (See: https://trinity-client.readthedocs.io/en/latest/contributing.html#pull-requests)

#### Cute Animal Picture

pexels-photo-1245356

### CODE REVIEW

**Comment by hwwhww** [comment]
There's a spec change of `verify_bitfield` (changed to `validate_bitfield` in our codebase) logic, pending on: https://github.com/ethereum/eth2.0-specs/pull/573. Otherwise, ready to review.

File: tests/eth2/beacon/test_helpers.py [hunk 1]
```diff
@@ -1050,24 +1051,28 @@ def test_generate_aggregate_pubkeys(activated_genesis_validators,
                                     sample_slashable_attestation_params,
                                     data):
     max_value_for_list = len(activated_genesis_validators) - 1
-    (indices, some_index) = _list_and_index(
+    (validator_indices, some_index) = _list_and_index(
```
Comment (NIC619) [inline tests/eth2/beacon/test_helpers.py]: It seems to me that the each elements  in `validator_indices` generated by `_list_and_index` should be unique? If this is the case, should we set `unique` to `True` in `st.lists(…)` in `_list_and_index` function?
`xs = data.draw(st.lists(elements, max_size=max_size, unique=True))`

I play around the test a little bit and found that when there are same index set in both `custody_bit_0_indices` and `custody_bit_1_indices` no errors were thrown.
Do you think we should check that there's no overlapping between `custody_bit_0_indices` and `custody_bit_1_indices`?

File: tests/eth2/beacon/test_helpers.py [hunk 2]
```diff
@@ -1123,35 +1112,55 @@ def _correct_slashable_attestation_params(
...
         fork,
         slot_to_epoch(params["data"].slot, epoch_length),
     )
-    valid_params[key] = poc_1_indices
 
-    signatures = poc_0_signatures + poc_1_signatures
+    valid_params["validator_indices"] = validator_indices
+    valid_params["custody_bitfield"] = get_empty_bitfield(len(validator_indices))
+
+    signatures = signatures
```
Comment (NIC619) [inline tests/eth2/beacon/test_helpers.py]: typo?

File: tests/eth2/beacon/test_helpers.py [hunk 3]
```diff
@@ -1123,35 +1112,55 @@ def _correct_slashable_attestation_params(
...
+    corrupt_custody_bitfield = get_empty_bitfield(validator_indices_length)
+    corrupt_custody_bitfield = set_voted(corrupt_custody_bitfield, 0)
+    return assoc(params, "custody_bitfield", corrupt_custody_bitfield)
+
+
+def _corrupt_validator_indices(params):
+    corrupt_validator_indices = (
+        params["validator_indices"][1],
+        params["validator_indices"][0],
+    ) + tuple(params["validator_indices"][1:])
```
Comment (NIC619) [inline tests/eth2/beacon/test_helpers.py]: This seems to result in an extra index, is this intended?
```suggestion
    ) + tuple(params["validator_indices"][2:])
```

File: tests/eth2/beacon/test_helpers.py (hunk 1 above)
Comment (hwwhww) [reply tests/eth2/beacon/test_helpers.py]: Yes. Nice catch! 👍

**Pull request merged**